synapse/config/server.py

```diff
@@ -1291,7 +1294,9 @@ def is_threepid_reserved(reserved_threepids, threepid):
     return False
 
 
-def read_gc_thresholds(thresholds):
+def read_gc_thresholds(
+    thresholds: Optional[List[Any]],
```

Comment:
Maybe `List[SupportsInt]` if you're feeling fancy!

Reply:
Also, below on +1314, looks like `listener` could be `Dict: [str, Any]`

Reply:
I'm not sure. That's what is *expected* by these methods, but not what is necessarily passed in?

Reply:
I'm not really sure what the best practice is here, but this is where we're taking any YAML junk and ensuring it is correct, so I think it would be a lie to say it has to accept integers?

Reply:
Agreed. (The argument I had in mind was that mypy could detect type errors if we described the intended inputs. But it's not going to be able to do that when the yaml comes at runtime.)